lib/gakubuchi/template.rb

```diff
@@ -2,54 +2,73 @@ module Gakubuchi
   class Template
     extend ::Forwardable
 
-    attr_reader :pathname
-    def_delegators :@pathname, :hash
+    attr_reader :source_path
+    def_delegator :source_path, :hash
 
     %w(== === eql?).each do |method_name|
       define_method(method_name) do |other|
-        self.class == other.class && @pathname.__send__(method_name, other.pathname)
+        self.class == other.class && source_path.public_send(method_name, other.source_path)
       end
     end
 
     def self.all
-      ::Dir.glob(root.join('**/*.html*')).map { |path| new(path) }
+      ::Dir.glob(root.join('**/*.html*')).map { |source_path| new(source_path) }
     end
 
     def self.root
-      ::Rails.root.join('app/assets', Gakubuchi.configuration.template_directory)
+      ::Rails.root.join('app/assets', ::Gakubuchi.configuration.template_directory)
     end
 
-    def initialize(path)
-      @pathname = ::Pathname.new(path)
+    def initialize(source_path)
+      path = ::Pathname.new(source_path)
+
+      case
+      when path.relative?
+        @source_path = self.class.root.join(path)
+      when path.ascend { |parent| break true if parent == self.class.root }
+        @source_path = path
+      else
+        fail "source path must include #{self.class.root} when it is absolute"
+      end
     end
 
-    def destination_pathname
-      dirname = relative_pathname.dirname
-      ::Rails.public_path.join(dirname, "#{relative_pathname.basename(extname)}.html")
+    def destination_path
+      ::Rails.public_path.join(logical_path)
+    end
+
+    def digest_path
+      resolved_path = view_context.asset_digest_path(logical_path.to_s)
+      return if resolved_path.nil?
+
+      Pathname.new(File.join(::Rails.public_path, view_context.assets_prefix, resolved_path))
     end
 
     def extname
       extnames = []
-      basename_without_ext = pathname.basename
+      basename = source_path.basename
 
       loop do
-        extname = basename_without_ext.extname
+        extname = basename.extname
         break if extname.empty?
 
         extnames.unshift(extname)
-        basename_without_ext = basename_without_ext.basename(extname)
+        basename = basename.basename(extname)
       end
 
       extnames.join
     end
 
-    def precompiled_pathname
-      asset = ::Rails.application.assets.find_asset(relative_pathname)
-      ::Rails.public_path.join('assets', asset.digest_path) if asset
+    private
+
+    def logical_path
+      dirname = source_path.relative_path_from(self.class.root).dirname
+      basename_without_ext = source_path.basename(extname)
+
+      ::Pathname.new(dirname).join("#{basename_without_ext}.html")
     end
 
-    def relative_pathname
-      pathname.relative_path_from(self.class.root)
+    def view_context
+      @view_context ||= ::ActionView::Base.new
```

Comment:
I decided to leave it because I found controllers of Rails creates a new instance every time.
cf. https://github.com/rails/rails/blob/7f18ea14c893cb5c9f04d4fda9661126758332b5/actionview/lib/action_view/rendering.rb#L71-L73